core/common/src/main/java/alluxio/PropertyKey.java

```diff
@@ -37,42 +37,162 @@
   /** A map from default property key's string name to the key. */
   private static final Map<PropertyKey, Object> DEFAULT_VALUES = new HashMap<>();
 
+  /**
+   * Builder to create {@link PropertyKey} instances.
+   */
+  public static final class Builder {
+    private String[] mAlias;
+    private Object mDefaultValue;
+    private String mDescription;
```

Comment:
this is added intentionally to unblock https://alluxio.atlassian.net/browse/ALLUXIO-2889